arch/Kconfig

```
@@ -994,3 +996,10 @@ config TOOLCHAIN_HAS_BUILTIN_FFS
 	default y if !(64BIT && RISCV)
 	help
 	  Hidden option to signal that toolchain has __builtin_ffs*().
+
+config ARCH_HAS_MEMORY_BARRIER
+	bool "Enable architecture specific memory barrier support"
+	help
+	  This option enable architecture specific function for
+	  the memory barrier API support. These functions can be
+	  implemented or builtin toolchain.
```

The `ARCH_HAS_*` symbols are usually hidden with no prompt.
```suggestion
	bool
	help
	  This option signifies that the architecture does support and implement the 
	  memory barrier functions.
```

---

Ack